packages/gestalt/src/contexts/NestingProvider.js

```diff
@@ -0,0 +1,41 @@
+// @flow strict
+import { type Context, type Element, type Node, useContext, createContext } from 'react';
+
+type NestingContextType = {|
+  nestedLevel: number,
+|};
+
+type Props = {|
+  children: Node,
+|};
+
+const NestingContext: Context<NestingContextType> = createContext<NestingContextType>({
+  nestedLevel: 0,
+});
+
+const { Provider } = NestingContext;
+
+function NestingProvider({ children }: Props): Element<typeof Provider> {
+  const { nestedLevel } = useContext(NestingContext);
+
+  const nextNestedLevel = (nestedLevel ?? 0) + 1;
```

Comment:
If I'm reading this logic correctly, we can only ever have nested levels of `1 | 2`? If `nestedLevel` doesn't exist, we start with 0, add 1 to it, then pass down the result. Can a component ever read the context and get `0`?